uw-frame-components/portal/main/controllers.js

```diff
@@ -73,17 +73,31 @@ define(['angular','require'], function(angular, require) {
   }]);
 
   /* Header */
-  app.controller('PortalHeaderController', ['$rootScope', '$scope','$location', 'NAMES', 'APP_FLAGS', 'MISC_URLS', function($rootScope, $scope, $location, NAMES, APP_FLAGS, MISC_URLS) {
+  app.controller('PortalHeaderController', ['$rootScope', '$scope','$location', 'NAMES', 'APP_FLAGS', 'MISC_URLS','notificationsService', function($rootScope, $scope, $location, NAMES, APP_FLAGS, MISC_URLS, notificationsService) {
     this.navbarCollapsed = true;
     this.showLogout = true;
     $scope.showSearch = false;
     $scope.showSearchFocus = false;
     $scope.APP_FLAGS = APP_FLAGS;
     $scope.MISC_URLS = MISC_URLS;
-
+    notificationsService.getAllNotifications().then(function(data) {
+      var notifications = data;
+      var priorityNotifications = [];
+      for( var i=0; i < notifications.length; i++ ) {
```

Comment:
Hmmm, thinking about this more, I'm saddened we have to call the service twice just to get a `boolean`. We do pass the `headerCtrl` into the notification directive, should we just use that?

Also, we would only want to run this function if notifications was enabled.